Sources/KlaviyoSwift/StateManagement.swift

```diff
@@ -323,21 +336,50 @@ struct KlaviyoReducer: ReducerProtocol {
         case let .enqueueProfile(profile):
             guard case .initialized = state.initalizationState,
                   let apiKey = state.apiKey,
-                  let anonymousId = state.anonymousId
+                  let _ = state.anonymousId
             else {
                 state.pendingRequests.append(.profile(profile))
                 return .none
             }
+            let pushTokenData = state.pushTokenData
             state.reset()
             state.updateStateWithProfile(profile: profile)
-            let request = KlaviyoAPI.KlaviyoRequest(
-                apiKey: apiKey,
-                endpoint: .createProfile(.init(data: .init(profile: profile, anonymousId: anonymousId))))
+            guard let anonymousId = state.anonymousId else {
+                return .none
+            }
+            if let tokenData = pushTokenData {
+                let request = KlaviyoAPI.KlaviyoRequest(
+                    apiKey: apiKey,
+                    endpoint: .registerPushToken(.init(pushToken: tokenData.pushToken, enablement: tokenData.pushEnablement.rawValue, background: tokenData.pushBackground.rawValue, profile: profile, anonymousId: anonymousId)))
 
-            state.queue.append(request)
+                state.enqueueRequest(request: request)
+            } else {
+                let request = KlaviyoAPI.KlaviyoRequest(
+                    apiKey: apiKey,
+                    endpoint: .createProfile(.init(data: .init(profile: profile, anonymousId: anonymousId))))
+
+                state.enqueueRequest(request: request)
+            }
             return .none
         case .resetProfile:
+            let pushTokenData = state.pushTokenData
             state.reset()
+            guard case .initialized = state.initalizationState,
+                  let apiKey = state.apiKey,
+                  let _ = state.anonymousId
+            else {
+                return .none
+            }
+            if let tokenData = pushTokenData {
+                guard let anonymousId = state.anonymousId else {
```

Comment:
Is this redundant on the same code on line 369?

Reply:
Yup good point. We can eliminate the first check now I thinkl